### ISSUE TITLE
Question about ordinary python code interreaction with theano code?

### ISSUE THREAD

**Issue by lfwin** [Theano/Theano#3413]
I try to implement a CNN modified code from CNN in Theano tutorial, but right now i want to change the learning rate like this 
```
    learning_rate = 0.01
    ...
    train_model = theano.function(inputs = [index], outputs = [cost,train_loss], 
          updates=gradient_updates_momentum(cost, params, learning_rate),
          givens={
            x: train_set_x[index * batch_size: (index + 1) * batch_size],
            y: train_set_y[index * batch_size: (index + 1) * batch_size]})
   ...
    done_looping = False    
    while (epoch_count < n_epochs) and (not done_looping):
        epoch_count = epoch_count + 1
        
        if epoch_count >= 2:
            learning_rate = 0.001
        minibatch_count = 0           
        for minibatch_index in numpy.random.permutation(n_train_batches).astype('int32'):
            #print "minibatch_index = ", minibatch_index
            step = (epoch_count - 1) * n_train_batches + minibatch_count

            if step % 100 == 0:
                print 'training @ iter = ', step
            cost_ij, train_loss = train_model(minibatch_index)
            this_cost_history.append(cost_ij)
            this_epoch_train_loss_history.append(train_loss)
            
            minibatch_count += 1
```

this does not actually modified learning rate, maybe this is because i did not understand Theano and ordinary python code interreaction with each other.

**Comment by abergeron** [comment]
Please use the mailing list for such questions, this is not a bug report.

Also, the problem is that python values get embedded as constants in the theano graph.  If you want to change the value, either make it an input or a shared variable.
